flink-filesystems/flink-s3-fs-presto/src/test/java/org/apache/flink/fs/s3presto/MinioTestContainerTest.java

```diff
@@ -0,0 +1,117 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+/**
+ * {@code MinioTestContainerTest} tests some basic functionality provided by {@link
+ * MinioTestContainer}.
+ */
+public class MinioTestContainerTest {
+
+    private static final String DEFAULT_BUCKET_NAME = "test-bucket";
+
+    @RegisterExtension
+    private static final EachCallbackWrapper<TestContainerExtension<MinioTestContainer>>
+            MINIO_EXTENSION =
+                    new EachCallbackWrapper<>(
+                            new TestContainerExtension<>(
+                                    () -> new MinioTestContainer(DEFAULT_BUCKET_NAME)));
+
+    private static MinioTestContainer getTestContainer() {
+        return MINIO_EXTENSION.getCustomExtension().getTestContainer();
+    }
+
+    private static AmazonS3 getClient() {
+        return getTestContainer().getClient();
+    }
+
+    @Test
+    public void testBucketCreation() {
+        final String otherBucketName = "other-bucket";
+        final Bucket otherBucket = getClient().createBucket(otherBucketName);
+
+        assertThat(otherBucket).isNotNull();
+        assertThat(otherBucket).extracting(Bucket::getName).isEqualTo(otherBucketName);
+
+        assertThat(getClient().listBuckets())
+                .map(Bucket::getName)
+                .containsExactlyInAnyOrder(
+                        getTestContainer().getDefaultBucketName(), otherBucketName);
+    }
+
+    @Test
+    public void testPutObject() throws IOException {
+        final String bucketName = "other-bucket";
+
+        getClient().createBucket(bucketName);
+        final String objectId = "test-object";
+        final String content = "test content";
+        getClient().putObject(bucketName, objectId, content);
+
+        final BufferedReader reader =
+                new BufferedReader(
+                        new InputStreamReader(
+                                getClient().getObject(bucketName, objectId).getObjectContent()));
+        assertThat(reader.readLine()).isEqualTo(content);
+    }
+
+    @Test
+    public void testFlinkConfigurationInitialization() {
+        final Configuration config = new Configuration();
+        getTestContainer().updateConfigAccordingly(config);
+
+        assertThat(config.containsKey("s3.endpoint")).isTrue();
+        assertThat(config.containsKey("s3.path.style.access")).isTrue();
+        assertThat(config.containsKey("s3.access.key")).isTrue();
+        assertThat(config.containsKey("s3.secret.key")).isTrue();
+    }
+
+    @Test
+    public void testDefaultBucketCreation() {
+        final Configuration config = new Configuration();
+        getTestContainer().updateConfigAccordingly(config);
```

Comment:
is this required? If so, why isnt it for testPutObject?

Reply:
not required. correct